### DISCUSSION TITLE
How to configure maven lib updates in external file

### DISCUSSION THREAD

**Discussion by tloszabno** [renovatebot/renovate#11000] [Request Help] [answered]
Hello,

in the company, we use some sort of external files to do the configuration of deployment. There I have dependency:
`somefile.json`
```
    {
      "name": "newRelic",
      "agentVersion": "4.12.1"
    },
```

and `renovate.config`

```
...
"regexManagers": [
    {
      "fileMatch": [
        "somefile.json"
      ],
      "matchStrings": [
        "\"agentVersion\": \"(?<currentValue>.*?)\""
      ],
      "depNameTemplate": "com.newrelic.agent.java:newrelic-agent",
      "datasourceTemplate": "maven",
      "versioningTemplate": "maven"
    }
  ]
...
```
but it does not work. 

Any advice on how to configure the update of the lib in such a situation?

**Comment by rarkins** [comment, marked as answer]
You'll need double backslashes (`\\`) instead of single. See https://github.com/renovatebot/.github/blob/main/default.json for some examples of our own patterns

**Reply by tloszabno** [reply]
why? I do not want to use regexp when slash, just escape using the exact character I want - in my case it is `\"`  - IMO single `\` should work.

**Comment by viceice** [comment]
Please provide debug logs, there should be some hints about regex manager
